### DISCUSSION TITLE
Declaration error using SimpleStorage

### DISCUSSION THREAD

**Discussion by mart-cervants** [smartcontractkit/full-blockchain-solidity-course-py#1017] [Q&A] [answered]
Hello, I have this error when running StorageFactory.sol, any idea what's the problem?

from solidity:
contracts/StorageFactory.sol:19:80: DeclarationError: Undeclared identifier.
        SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex]));
                                                                               ^-----------------^

and the program is the following:

// SPDX-License-Identifier: MIT

pragma solidity ^0.6.0;

import "./SimpleStorage.sol";

contract StorageFactory is SimpleStorage{

    SimpleStorage[] public simpleStorageArray;

    function createSimpleStorageContract() public {
        SimpleStorage simpleStorage = new SimpleStorage();
        simpleStorageArray.push(simpleStorage);
    }

    function sfStore(uint256 _simpleStorage, uint256 _simpleStorageNumber) public {
        // Address
        // ABI
        SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])); 
        simpleStorage.store(_simpleStorageNumber);

        // Simplified
        //SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).store(_simpleStorageNumber);
    }

    function sfGet(uint256 _simpleStorageIndex) public view returns (uint256) {
        SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex]));
        return simpleStorage.retrieve();

        // Simplified
        //return SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).simpleStorage.retrieve();
    }
}

**Comment by n4n0b1t3** [comment, marked as answer]
in sfStore you are using the function parameter _simpleStorageNumber and not _simpleStorageIndex

in case you are using remix there should be a red marker on the left side of the editor pane at the height of this line. Hover with your mouse over it and you will see the error.

**Reply by mart-cervants** [reply]
thanks, you're right .

I didnt see that.
